lib/OpenQA/WebAPI/Auth/OpenID.pm

```diff
@@ -108,18 +108,22 @@ sub auth_response ($c) {
     };
 
     $csr->handle_server_response(
-        not_openid =>
-          sub () { $err_handler->('Failed to login', 'OpenID provider returned invalid data. Please retry again') },
+        not_openid => sub () {
+            my $txt
+              = 'Likely, OpenID message is not considered a valid object. It is missing, at least, "openqa.mode" value';
```

Comment:
I don't understand the "Likely". What do you mean by that?

Reply:
the `not_openid` is set undef in different conditions. In the `IndirectMessage.pm` you will find that return undef in a single `$self->get('mode')` check. Then in other places checks other keys of the object. To me this says that the object might be just incomplete for some reason, and based on the keys the object is assumed invalid. But the problem might also occur if there is a problem on the OpenID Provider respond or the URI is wrong.

Reply:
Ok, but regardless the handler is seeing invalid data and can not process the request, right? So I don't think the "Likely" helps. And the second part is also adding details which can be considered good but then it does not really help with the issue that the data can not be processed. Imho the more important part that should be made explicit is what the effect of the situation is, i.e. you explained the "what". Still missing is the "so what"

Reply:
Could be `so what` a prompt to retry but we remove it because this is not useful to the end user. I can think only a suggestion to check the OP. or a link to the documentation. 
The effect is not disastrous and it is not happen that often. Not sure what info makes sense

Reply:
But this text is also provided to users. What should it tell them?

Reply:
*login failed*

Reply:
@b10n1k Did you look into providing more details based on [Net::OpenID::Consumer](https://metacpan.org/pod/Net::OpenID::Consumer) as was suggested [in the original question about this message](https://github.com/os-autoinst/openQA/pull/5952#discussion_r1777294685)?